static/app/components/modals/navigateToExternalLinkModal.tsx

```diff
@@ -0,0 +1,80 @@
+import {Fragment} from 'react';
+import styled from '@emotion/styled';
+
+import {ModalRenderProps} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import {IconWarning} from 'sentry/icons';
+import {t} from 'sentry/locale';
+
+type Props = ModalRenderProps & {
+  linkText: string;
+};
+
+function NavigateToExternalLinkModal({Body, closeModal, Header, linkText = ''}: Props) {
+  return (
+    <Fragment>
+      <Header closeButton>
+        <HeaderContainer>
+          <IconWarning size="sm" color="warningText" />
+          <h2>{t('Leaving Sentry Warning')}</h2>
+        </HeaderContainer>
+      </Header>
+      <Body>
+        <p>
+          {t(
+            'You are now leaving Sentry and will be redirected to the following external website'
+          )}
+        </p>
+        <p>{linkText}</p>
+        <p />
+      </Body>
+      <ButtonContainer>
+        <ButtonBar>
+          <Button
+            priority="primary"
+            onClick={() => {
+              window.open(linkText, '_blank', 'noreferrer');
```

Comment:
Instead of onClick, you can use add these props: `href={link}` and `external={true}`